sources/web/datalab/polymer/components/datalab-sessions/datalab-sessions.ts

```diff
@@ -113,6 +113,26 @@ class SessionsElement extends Polymer.Element {
       });
   }
 
+  /**
+   * Calls the ApiManager to terminate the selected sessions.
+   */
+  _shutdownSelectedSessions() {
+    const selectedIndices = this.$.sessions.getSelectedIndices();
+    if (selectedIndices.length) {
+      let shutdownPromises = selectedIndices.map((i: number) => {
+        return ApiManager.shutdownSessionAsync(this._sessionList[i].id);
+      });
+
+      // TODO: [yebrahim] If at least one delete completes then a failure happens with
+      // any of the rest, _fetchFileList will never be called.
```

Comment:
Do you mean _fetchSessionList will never be called?
The 'then' is only called after all promises succeed; if any of them fail, the all() fails immediately. "At least one ... completes" is not part of the calculation.